File: app/algorithm/AA_25-26/SchedulingAPI/Models/TimePreference.cs
namespace SchedulingAPI.Models
{
    public class TimePreference
    {
        public int Start { get; set; }
        public int End { get; set; }
        public string PreferredGenre { get; set; } = string.Empty;
        public int Bonus { get; set; }
    }
}

File: app/algorithm/AA_25-26/models/time_preference.py
class TimePreference:
    def __init__(self, start, end, preferred_genre, bonus):
        self.start = start
        self.end = end
        self.preferred_genre = preferred_genre
        self.bonus = bonus

    def __repr__(self):
        return f"TimePreference({self.start}-{self.end}, Genre: {self.preferred_genre}, Bonus: {self.bonus})"